src/glue/jobs/json_s3_to_parquet.py

```diff
@@ -132,105 +141,246 @@ def add_index_to_table(table_key, table_name, processed_tables, unprocessed_tabl
-                     "month", "day", "recordid"])
-                .distinct())
+        parent_index = parent_table.select(
+            [original_field_name, "assessmentid", "year", "month", "day", "recordid"]
+        ).distinct()
         this_index = parent_index.withColumnRenamed(original_field_name, "id")
-        df_with_index = this_table.join(
-                this_index,
-                on = "id",
-                how = "inner")
+        df_with_index = this_table.join(this_index, on="id", how="inner")
         # remove prefix from field names
         field_prefix = table_key.replace(f"{table_name}_", "") + ".val."
         columns = list(df_with_index.columns)
         for c in columns:
             # do nothing if c is id, index, or partition field
-            if f"{original_field_name}.val" == c: # field is an array
+            if f"{original_field_name}.val" == c:  # field is an array
                 succinct_name = c.replace(".", "_")
-                df_with_index = df_with_index.withColumnRenamed(
-                        c, succinct_name)
+                df_with_index = df_with_index.withColumnRenamed(c, succinct_name)
             elif field_prefix in c:
                 succinct_name = c.replace(field_prefix, "").replace(".", "_")
                 # If key is a duplicate we keep the original field name
                 if succinct_name in df_with_index.columns:
                     continue
-                df_with_index = df_with_index.withColumnRenamed(
-                        c, succinct_name)
+                df_with_index = df_with_index.withColumnRenamed(c, succinct_name)
     return df_with_index
 
+
+def cast_glue_data_types(
+    dynamic_frame: DynamicFrame,
+    glue_database_name: str,
+    glue_table_name: str,
+) -> DynamicFrame:
+    """Takes a DynamicFrame and Glue table name and force Glue to respect our data
+    type choices when resolving types from a DynamicFrame object to a Spark Dataframe.
+    This
+
+    Args:
+        dynamic_frame (awsglue.DynamicFrame): DynamicFrame object input
+        glue_database_name (str): Name of the database to pull the glue table from
+        glue_table_name (str): Name of the glue table to pull the json schema from
+
+    Returns:
+        awsglue.DynamicFrame: returns a DynamicFrame with those resolved data types
+    """
+    # Load the input glue data
+    glue_client = boto3.client("glue")
+    response = glue_client.get_table(
+        DatabaseName=glue_database_name, Name=glue_table_name
+    )
+    table = response["Table"]
+    glue_table_schema = table["StorageDescriptor"]["Columns"]
+    specs = []
+    for field_info in glue_table_schema:
+        json_schema = parse_hive_schema(
+            hive_str=field_info["Type"], top_level_field=field_info["Name"]
+        )
+        specs.extend(convert_json_schema_to_specs(json_schema))
+    glue_table = dynamic_frame.resolveChoice(specs=specs)
+    return glue_table
+
+
```

Comment:
Nit: is there a python package that does this already?

Reply:
Are you referring to `parse_hive_schema`? I did look for one and as a result of my research it sounds like there wasn't anything that does this (converting from json hive schema string to dictionary format) out of the box and it has to be manually done in this way. I tried looking into some of the custom parsing libraries like `pyparsing` but they didn't seem to support this use case.